upgrade/steps.go

```diff
@@ -805,3 +805,54 @@ func ReplaceAssertNoError(ctx Context, repo ProviderRepo) (step.Step, error) {
 	}
 	return step.Combined("Replace AssertNoError with AssertNoErrorf", steps...), nil
 }
+
+// UpgradePulumiVersions reads the current Pulumi SDK version from go.mod and applies it to:
+// sdk/go.mod
+// examples/go.mod - we also infer the `pkg` version here and add it.
+func BridgePulumiVersions(ctx Context, repo ProviderRepo) (step.Step, error) {
```

Comment:
I don't see where `BridgePulumiVersions` can return an error.
```suggestion
func BridgePulumiVersions(ctx Context, repo ProviderRepo) step.Step {
```

Reply:
I love moving this concept into a function.

Reply:
It's returning an error here; https://github.com/pulumi/upgrade-provider/pull/152/files#diff-d5c54fe7fdac4cea34363fc36154d215683b9b55e14428dd7cd578204f6c8d92R820